### PR TITLE
Update meson to 0.60.2

### PR DESCRIPTION
**Pull request by Neumann-A**
- rename tool-meson to vcpkg-tool-meson. (deprecate old port or to be more precise use for testing of `vcpkg_find_acquire_program(MESON)`)
- make ports explicitly depend on vcpkg-tool-meson if they require meson.
- make it so that no future changes to `vcpkg_find_acquire_program` are necessary if meson gets updated. (Will reduce the number of CI rebuilds.)

- [x] depends #21673 

I expect ports to break with this because meson updates break stuff even more than cmake updates

### CODE REVIEW

File: ports/gstreamer/vcpkg.json [hunk 1]
```diff
@@ -34,7 +34,14 @@
...
     "openssl",
-    "tool-meson",
+    {
+      "name": "vcpkg-tool-meson",
+      "host": true
+    },
+    {
+      "name": "vcpkg-tool-meson",
+      "host": true
+    },
```
Comment (Neumann-A) [inline ports/gstreamer/vcpkg.json:44]:
```suggestion
```

accidental duplication

File: scripts/cmake/vcpkg_find_acquire_program.cmake [hunk 2]
```diff
@@ -88,17 +88,19 @@ function(z_vcpkg_find_acquire_program_find_external program)
...
+            set("${program}" "" CACHE INTERNAL "")
         endif()
-        unset(SCRIPT_${program} CACHE)
+        unset(SCRIPT_${arg_PROGRAM_NAME} CACHE)
     endif()
 
     if("${version_command}" STREQUAL "")
         set(version_is_good ON) # can't check for the version being good, so assume it is
-    else()
+    elseif(${program}) # onyl do a version check if ${program} has a value
```
Comment (c72578) [inline scripts/cmake/vcpkg_find_acquire_program.cmake:103]: onyl->only

**Comment by Neumann-A** [comment]
@BillyONeal  can you say something about the v8 failure. Is that a baseline failure?

**Comment by Neumann-A** [comment]
opened https://github.com/mesonbuild/meson/issues/9618 because they regressed my previous glib fix for libintl.

**Comment by BillyONeal** [comment]
> @BillyONeal can you say something about the v8 failure. Is that a baseline failure?
> 
> probably baseline see #21620

Indeed; I was forced to engage Twitter Driven Development https://twitter.com/sokac/status/1463182302679220230

**Comment by BillyONeal** [comment]
I believe Google folks fixed v8

**Comment by Neumann-A** [comment]
@BillyONeal:
hpx on linux:
`-- Could NOT find TCMalloc (missing: TCMALLOC_LIBRARY TCMALLOC_INCLUDE_DIR) `
```
CMake Error at cmake/HPX_Message.cmake:51 (message):
  HPX_WITH_MALLOC was set to tcmalloc, but tcmalloc could not be found.
  Valid options for HPX_WITH_MALLOC are: system, tcmalloc, jemalloc,
  mimalloc, tbbmalloc, and custom
Call Stack (most recent call first):
  cmake/HPX_SetupAllocator.cmake:47 (hpx_error)
  CMakeLists.txt:2166 (include)
```

looks like a baseline regression?

**Comment by BillyONeal** [comment]
> @BillyONeal: hpx on linux: `-- Could NOT find TCMalloc (missing: TCMALLOC_LIBRARY TCMALLOC_INCLUDE_DIR) `
> 
> ```
> CMake Error at cmake/HPX_Message.cmake:51 (message):
>   HPX_WITH_MALLOC was set to tcmalloc, but tcmalloc could not be found.
>   Valid options for HPX_WITH_MALLOC are: system, tcmalloc, jemalloc,
>   mimalloc, tbbmalloc, and custom
> Call Stack (most recent call first):
>   cmake/HPX_SetupAllocator.cmake:47 (hpx_error)
>   CMakeLists.txt:2166 (include)
> ```
> 
> looks like a baseline regression?

We did just remove hpx from ci.baseline.txt here but that was because it passed in CI (it got PASSING: REMOVE FROM FAIL LIST). Perhaps it's order dependent?

**Comment by BillyONeal** [comment]
I'm able to repro the hpx failure locally, stand by...

**Comment by BillyONeal** [comment]
https://github.com/microsoft/vcpkg/pull/21673

**Comment by Neumann-A** [comment]
the gstreamer regression also seems like baseline:
`LINK : fatal error LNK1181: cannot open input file 'm.lib'`
```
, {"name": "flac"
, "version": "1.3.3"
, "compile_args": ["-ID:/installed/x64-windows/include"]
, "link_args": ["D:/installed/x64-windows/lib/FLAC.lib"
, "D:/installed/x64-windows/lib/ogg.lib"
, "m.lib"]}
```
```
gst-plugins-good| Dependency flac found: NO unknown version, but need: ['>=1.1.4']
gst-plugins-good| Run-time dependency flac found: YES 1.3.3
```
so its an issue in libflac. fixed it manually.

**Comment by JackBoosY** [comment]
The hpx regression should not related to this PR.

**Comment by BillyONeal** [comment]
The osx failure ran on the one we suspect is unhealthy from https://github.com/microsoft/vcpkg/pull/21676#issuecomment-982010390 but I'm not resubmitting because it looks like there are version DB updates needed. I've removed -06 from the fleet.

**Comment by Neumann-A** [comment]
hmm the formater in vcpkg-tool changed?

**Comment by BillyONeal** [comment]
> hmm the formater in vcpkg-tool changed?

https://github.com/microsoft/vcpkg-tool/pull/282 may have had an effect, yes

File: scripts/cmake/vcpkg_find_acquire_program.cmake [hunk 3]
```diff
@@ -88,17 +88,19 @@ function(z_vcpkg_find_acquire_program_find_external program)
...
+            set("${program}" "" CACHE INTERNAL "")
         endif()
-        unset(SCRIPT_${program} CACHE)
+        unset(SCRIPT_${arg_PROGRAM_NAME} CACHE)
     endif()
 
     if("${version_command}" STREQUAL "")
         set(version_is_good ON) # can't check for the version being good, so assume it is
-    else()
+    elseif(${program}) # only do a version check if ${program} has a value
```
Comment (JackBoosY) [inline scripts/cmake/vcpkg_find_acquire_program.cmake:103]: What if NOT `"${version_command}" STREQUAL ""` and NOT `${program}` ?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
I'm merging this despite the qwt-qt6 failure because that port is currently broken and being backed out https://github.com/microsoft/vcpkg/pull/21825

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**